### PR TITLE
Feat/enable argo notification

### PR DESCRIPTION
**Pull request by FaGru3n**
- generally add notification secrets
- enabled teams urls to devsecops-testing cluster 
- also added private demo channel for testing purposes

already updated our vault secret to 2 entries 

my idea was general enablement for notification, because updating argo internal ressoucres from repo product-onboarding 
seems not working with current kustomization strategy.
https://github.com/kubernetes-sigs/kustomize/blob/master/examples/patchMultipleObjects.md was tested on product-onboarding repository

For this approach we need to add may product specifica in our argo base repo but with this, we have may multible channels only based on the subspribed apps we can declare afterwards

For testing added a demo-channel in devsecops cluster

### CODE REVIEW

**Comment by FaGru3n** [comment]
Hint to test the overlays with secrets and configmap it this pr is only capable for devsecpostesting

only thing that will added in base is the generall demo channel for the cluster

**Comment by FaGru3n** [comment]
after 'kustomize build devsecops-testing' 

results are:
``` yaml
apiVersion: v1
kind: Secret
metadata:
  name: argocd-notifications-secret
  namespace: argocd
stringData:
  demo-teams-channel-url: <path:devsecops/data/apps/argocd/argocd-notifications-secret#demo-teams-channel-url>
  irs-teams-channel-url: <path:devsecops/data/apps/argocd/argocd-notifications-secret#irs-teams-channel-url>
type: Opaque
```

``` yaml
apiVersion: v1
data:
  service.teams.irs: |
    recipientUrls:
      irs-dev-teams: $irs-teams-channel-url
      demo-teams-channel: $demo-teams-channel-url
  template.app-created: |
    email:
      subject: Application {{.app.metadata.name}} has been created.
    message: Application {{.app.metadata.name}} has been created.
    teams:
      title: Application {{.app.metadata.name}} has been created.
...
kind: ConfigMap
metadata:
  name: argocd-notifications-cm
  namespace: argocd

**Review by SebastianBezold — APPROVED** [review]

File: apps/argocd/base/argocd-notifications-secrets.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: argocd-notifications-secret
+type: Opaque
+stringData:
+  demo-teams-channel-url: <path:devsecops/data/apps/argocd/argocd-notifications-secret#demo-teams-channel-url>
```
Comment (carslen) [inline apps/argocd/base/argocd-notifications-secrets.yaml:7]: Why is this secret/channel-url added to all clusters via `apps/argocd/base`?
Please use a consortia MS Teams channel for demo purpose but not a Mercedes-Benz Teams channel.

File: apps/argocd/overlays/devsecops-testing/secrets/notification-secret-irs.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
```
[the remark this answers is not in the document]
Comment (FaGru3n) [reply apps/argocd/overlays/devsecops-testing/secrets/notification-secret-irs.yaml]: https://github.com/catenax-ng/k8s-cluster-stack/pull/353/commits/dbede94f6c774326e57e1548ad8e72165d00a6f9

File: apps/argocd/base/argocd-notifications-secrets.yaml (hunk 1 above)
Comment (FaGru3n) [reply apps/argocd/base/argocd-notifications-secrets.yaml:7]: already asked @Siegfriedk  and @COROANN who can help for an Cat-X Channel where we can add an webhook

**Review by carslen — APPROVED** [review]
LGTM

**Pull request merged**